File: _sources/frontend/vuejs/src/models/Product.ts
import { PriceItem } from './PriceItem';
import { ProductUsageType } from './ProductUsageType';

export interface Product {
  id: string;
  name: string;
  description: string | null;
  image: string | null;
  labels: string[] | null;
  tags: string[] | null;
  prices: PriceItem[] | null;
  usages: ProductUsageType[] | null;
  dateCreated: string;
  dateModified: string;
}

export function createDefaultProduct(): Product {
  return {
    id: '',
    name: '',
    description: null,
    image: null,
    labels: [],
    tags: [],
    prices: [],
    usages: [],
    dateCreated: new Date().toISOString(),
    dateModified: new Date().toISOString(),
  };
}

File: _sources/backend/api/ChefMate.API/Models/Product.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace ChefMate.API.Models;

[BsonIgnoreExtraElements]
public class Product : ModelBase
{
    public required string Name { get; set; }
    public List<string>? Labels { get; set; }
    public string? Type { get; set; }
    public string? Comments { get; set; }
    public string? Unit { get; set; }
    public List<PriceItem>? Prices { get; set; }
    public List<string>? Tags { get; set; }
    public string? ProfileId { get; set; }
    public double? QuantityPerMonth { get; set; }
}